### PR TITLE
e2e_node: Disable auto restart on CoreOS.

### LINKED ISSUE #24423: kubelet test flake - coreos killed ssh connection mid test
>
17:20:21 • [SLOW TEST:11.682 seconds]
17:20:21 PrivilegedPod
17:20:21 /var/lib/jenkins/workspace/kubelet-gce-e2e-ci/go/src/k8s.io/kubernetes/test/e2e_node/privileged_test.go:84
17:20:21   should test privileged pod
17:20:21   /var/lib/jenkins/workspace/kubelet-gce-e2e-ci/go/src/k8s.io/kubernetes/test/e2e_node/privileged_test.go:83
17:20:21 ------------------------------
17:20:21 I0418 00:19:33.350505    1028 privileged_test.go:154] Deserialized output is map[error:exit status 2 output:RTNETLINK answers: Operation not permitted
17:20:21 ]
17:20:21 ••••••Connection to tmp-node-e2e-ca7b49a2-e2e-node-coreos-stable20160218-image closed by remote host.
17:20:21 
17:20:21 ================================================================

[full logs](https://console.cloud.google.com/storage/browser/kubernetes-jenkins/logs/kubelet-gce-e2e-ci/1654)

### PR DESCRIPTION
**Pull request by yifan-gu**
Fix #24423 

cc @crawford @kubernetes/sig-testing

### CODE REVIEW

**Comment by yifan-gu** [comment]
cc @pwittrock

**Comment by pwittrock** [comment]
@yifan-gu Looks like masking locksmithd failed:

>
F0430 01:48:52.180059     931 e2e_node_suite_test.go:135] Could not mask locksmithd: exit status 1

**Comment by yifan-gu** [comment]
Hmm, maybe need sudo?

**Comment by pwittrock** [comment]
Makes sense.  Want to update e2e_remote.go ln 158 with sudo **fmt.Sprintf("sudo ./e2e_node.test**
?

File: test/e2e_node/e2e_node_suite_test.go [hunk 1]
```diff
@@ -118,3 +124,16 @@ func (lr *LogReporter) SpecSuiteDidEnd(summary *types.SuiteSummary) {
...
+	data, err := ioutil.ReadFile("/etc/os-release")
+	if err != nil {
+		glog.Fatalf("Could not read /etc/os-release: %v", err)
+	}
+	if bytes.Contains(data, []byte("ID=coreos")) {
+		if output, err := exec.Command("sudo", "systemctl", "mask", "--now", "locksmithd").CombinedOutput(); err != nil {
+			glog.Fatalf("Could not mask locksmithd: %v, output: %q", err, string(output))
+		}
+	}
+	glog.Infof("Locksmithd is masked successfully")
```
Comment (yifan-gu) [inline test/e2e_node/e2e_node_suite_test.go]: @pwittrock Hmm, I didn't see this message in the log. So the function didn't seem to be called at all?

Comment (pwittrock) [inline test/e2e_node/e2e_node_suite_test.go]: I see it in the [node e2e logs](https://storage.cloud.google.com/kubernetes-jenkins/pr-logs/pull/25004/node-pull-build-e2e-test/3847/build-log.txt)

>
Success Finished Host tmp-node-e2e-54dd96d6-e2e-node-coreos-stable20160218-image Test Suite
Warning: Permanently added 'tmp-node-e2e-54dd96d6-e2e-node-coreos-stable20160218-image'
...
I0502 07:07:00.610062     911 e2e_node_suite_test.go:101] ******************************************************
I0502 07:07:00.774986     911 e2e_node_suite_test.go:138] Locksmithd is masked successfully

**Comment by pwittrock** [comment]
@k8s-bot integration test this issue #24952

**Comment by pwittrock** [comment]
@k8s-bot unit test this issue #24952

**Pull request merged**